src/realm/object-store/shared_realm.hpp

```diff
@@ -135,6 +135,10 @@ struct RealmConfig {
     {
         return schema_mode == SchemaMode::ReadOnly;
     }
+    bool is_schema_additive() const
+    {
+        return schema_mode == SchemaMode::AdditiveExplicit || schema_mode == SchemaMode::AdditiveDiscovered;
```

Comment:
I noticed in the code that we consider also `ReadOnly` mode like it was additive, do I need to add it in here as well?

Reply:
Yes, I believe we should also be able to open a subset of the schema in ReadOnly mode and have dynamic links work. If so, then this method should be renamed to something more general along the lines of "allows_subset". But let's get confirmation from an SDK on this. cc: @nirinchev @tgoyne

Reply:
That sounds reasonable to me

Reply:
OK, let's rename the method and give it another go.